llmfoundry/models/layers/norm.py

```diff
@@ -111,9 +112,55 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
                             self.eps).to(dtype=x.dtype)
 
 
+class TritonRMSNorm(torch.nn.Module):
+
+    def __init__(
+        self,
+        normalized_shape: Union[int, List[int], torch.Size],
+        eps: float = 1e-5,
+        device: Optional[torch.device] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        super().__init__()
+        self.eps = eps
+
+        # Flash Attention expect a flat tensor
+        if isinstance(normalized_shape, int):
+            hidden_size = normalized_shape
+        elif isinstance(normalized_shape, list):
+            hidden_size = math.prod(normalized_shape)
+        elif isinstance(normalized_shape, torch.Size):
+            hidden_size = math.prod(normalized_shape)
+
+        self.weight = torch.nn.Parameter(
+            torch.ones(hidden_size, device=device, dtype=dtype))
+        self.register_parameter('bias', None)
+
+        try:
```

Comment:
nit: move try/except to the top of the constructor and attr setting out of it, so its just the import that is try/excepted